backend/amd64/emit.ml

```diff
@@ -747,18 +788,59 @@ let emit_float_constant f lbl =
 
 let vec128_constants = ref ([] : (Cmm.vec128_bits * L.t) list)
 
+let vec256_constants = ref ([] : (Cmm.vec256_bits * L.t) list)
+
+let vec512_constants = ref ([] : (Cmm.vec512_bits * L.t) list)
+
 let add_vec128_constant bits =
   try List.assoc bits !vec128_constants
   with Not_found ->
     let lbl = L.create Sixteen_byte_literals in
     vec128_constants := (bits, lbl) :: !vec128_constants;
     lbl
 
-let emit_vec128_constant ({ high; low } : Cmm.vec128_bits) lbl =
+let add_vec256_constant bits =
+  try List.assoc bits !vec256_constants
+  with Not_found ->
+    let lbl = L.create Thirtytwo_byte_literals in
+    vec256_constants := (bits, lbl) :: !vec256_constants;
+    lbl
+
+let add_vec512_constant bits =
+  try List.assoc bits !vec512_constants
+  with Not_found ->
+    let lbl = L.create Sixtyfour_byte_literals in
+    vec512_constants := (bits, lbl) :: !vec512_constants;
+    lbl
```

Comment:
Is it worth factoring out the three
`add_vecXYZ_constant` functions?
(I am not positive it is.)